Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -603,20 +601,20 @@ Bytecode::CodeGenerationErrorOr<void> SuperCall::generate_bytecode(Bytecode::Gen
         // This generates a single argument, which will be implicitly passed in accumulator
         MUST(argument.value->generate_bytecode(generator));
     } else {
-        argument_registers.ensure_capacity(m_arguments.size());
-
-        for ([[maybe_unused]] auto const& arg : m_arguments) {
-            auto arg_reg = generator.allocate_register();
-            argument_registers.unchecked_append(arg_reg);
-        }
-        for (size_t i = 0; i < m_arguments.size(); ++i) {
-            TRY(m_arguments[i].value->generate_bytecode(generator));
-            generator.emit<Bytecode::Op::Store>(argument_registers[i]);
-        }
-        if (!argument_registers.is_empty())
-            generator.emit_with_extra_register_slots<Bytecode::Op::NewArray>(2, AK::Array { argument_registers.first(), argument_registers.last() });
-        else
+        if (!m_arguments.is_empty()) {
+            // FIXME: For some reason a spread Argument is not a SpreadExpression,
+            //        but Arguments have their own class having a boolean denoting spreads
```

Comment:
uhhh...allocating because of this seems a little excessive? I'd rather duplicate that code instead but either way is fine I guess

Reply:
Sadly the generation function is not nice, so
I guess we'll eat this cost for now, and maybe change CallExpressions arguments in the future to avoid this oddity